### ISSUE TITLE
Livepatch does not enable in latest [Azure] Ubuntu 18.04 Pro image

### ISSUE THREAD

**Issue by cmars** [canonical/ubuntu-advantage-client#1425]
Our CI caught this and I just spun up an instance to confirm it. It looks like the most recent Ubuntu 18.04 Pro image in Azure fails to install (or find) canonical-livepatch when trying to enable it on auto-attach.

/var/log/ubuntu-advantage.log shows (resource token redacted):
```
2021-03-03 19:02:40,902 - livepatch.py:(174) [DEBUG]: Livepatch not enabled. Invalid command specified '/snap/bin/canonical-livepatch status'.
2021-03-03 19:02:40,903 - util.py:(560) [DEBUG]: Writing file: /var/lib/ubuntu-advantage/status.json
2021-03-03 19:02:41,762 - cli.py:(659) [DEBUG]: Executed with sys.argv: ['/usr/bin/ua', 'status']
2021-03-03 19:02:41,762 - util.py:(376) [DEBUG]: Reading file: /var/lib/ubuntu-advantage/private/machine-token.json
...
2021-03-03 19:02:42,189 - livepatch.py:(174) [DEBUG]: Livepatch not enabled. Invalid command specified '/snap/bin/canonical-livepatch status'.
2021-03-03 19:02:42,189 - util.py:(560) [DEBUG]: Writing file: /var/lib/ubuntu-advantage/status.json
...
2021-03-03 19:05:14,663 - util.py:(488) [DEBUG]: Ran cmd: /usr/bin/snap install canonical-livepatch, rc: 0 stderr: b''
2021-03-03 19:05:14,966 - util.py:(488) [DEBUG]: Ran cmd: /snap/bin/canonical-livepatch config remote-server=https://livepatch.canonical.com, rc: 0 stderr: b''
2021-03-03 19:05:15,019 - livepatch.py:(174) [DEBUG]: Livepatch not enabled. Failed running command '/snap/bin/canonical-livepatch status' [exit(1)]. Message: 
2021-03-03 19:05:15,645 - util.py:(488) [DEBUG]: Ran cmd: /snap/bin/canonical-livepatch enable ***REDACTED***, rc: 0 stderr: b''
```

**Comment by blackboxsw** [comment]
@cmars thanks for the quick bug here, nothing in these logs suggests that livepatch didn't get setup.

TLDR: I think this is a case where ua auto-attach was still running on the system at the point when someone ran "ua status"

Sorry those logs are very verbose and, as a result,  confusing. I can see how we interpret them as a failure to setup livepatch.

Given the order of the logs attached above, we- can see `['/usr/bin/ua', 'status']` was run on this system before the logs that say livepatch was enabled. So, I'm not surprised an earlier call to `ua status` didn't show livepatch attached yet. Which would fail CI if checked while `ua auto-attach` is in progress.

This log is actually ua client determining if it needs to install canonical-livepatch:
   `Livepatch not enabled. Invalid command specified '/snap/bin/canonical-livepatch status'.`
Then the next line shows it installing livepatch.

In ua-client 25.0 or later we have a `ua status --wait` which will block until `ua auto-attach` is complete.

For CI testing on 24.4 or earlier, I'd say CI should wait on two things `cloud-init status --wait` and  `ps | grep -q auto-attach == 0`

**Comment by blackboxsw** [comment]
also to note. on Azure bionic ua client has to wait for snaps to seed, `snap install canonical-livepatch` and 
from systemd-analyze blame: 
    5min 15.842s ua-auto-attach.service

that 5 mins looks way too much like a timeout.
From logs on my own box I see :
```bash
2021-03-03 19:24:37,455 - util.py:(488) [DEBUG]: Ran cmd: /usr/bin/snap wait system seed.loaded, rc: 0 stderr: b''
....
2021-03-03 19:29:29,573 - util.py:(488) [DEBUG]: Ran cmd: /usr/bin/snap install canonical-livepatch, rc: 0 stderr: b''
```
nearly 5 mins between seed.loaded exit 0 and an attempt to snap install canonical-livepatch.
This is probably something we need to peek at separately.
